From b0b1af2f6f992bfeae1cc250e133ee9c673a9e33 Mon Sep 17 00:00:00 2001
From: Jonathan David <jonathan.david@ni.com>
Date: Fri, 1 Apr 2016 10:10:50 -0500
Subject: [PATCH] Add NI specific changes to VxWorks config file

- Enabled FTP protocol
- Disabled IPv6
- Disabled HAVE_ALLOCA_H
- Enabled HAVE_CLOCK_GETTIME_MONOTONIC
- Removed HAVE_DLFCN_H
- Disabled HAVE_FCNTL_O_NONBLOCK (CAR # 410563)
- Removed HAVE_GETADDRINFO (CAR # 232229)
- Disabled HAVE_GETHOSTBYADDR_R and HAVE_GETHOSTBYADDR_R_8
- Disabled HAVE_GETHOSTBYNAME (CAR # 232229)
- Removed HAVE_IO_H
- Removed HAVE_LIBZ and HAVE_ZLIB_H
- Removed HAVE_MALLOC_H
- Removed HAVE_MEMORY_H
- Removed HAVE_TERMIOS_H and HAVE_TERMIO_H
- Removed RANDOM_FILE
- Fixed recvfrom() configuration
- Fixed recv() configuration
- Fixed send() configuration
- Changed SIZEOF_OFF_T to 4
- Unset USE_OPENSSL; it is set by our MAK files

Upstream-Status: Inappropriate [configuration]
---
 lib/config-vxworks.h | 56 ++++++++++++++++++++++----------------------
 1 file changed, 28 insertions(+), 28 deletions(-)

diff --git a/lib/config-vxworks.h b/lib/config-vxworks.h
index 19f2d2354..4dd1c056e 100644
--- a/lib/config-vxworks.h
+++ b/lib/config-vxworks.h
@@ -45,7 +45,7 @@
 /* #undef CURL_DISABLE_FILE */
 
 /* to disable FTP */
-#define CURL_DISABLE_FTP 1
+/* #undef CURL_DISABLE_FTP */
 
 /* to disable HTTP */
 /* #undef CURL_DISABLE_HTTP */
@@ -81,7 +81,7 @@
 /* #undef EGD_SOCKET */
 
 /* Define if you want to enable IPv6 support */
-#define ENABLE_IPV6 1
+/* #undef ENABLE_IPV6 */
 
 /* Define to the type qualifier of arg 1 for getnameinfo. */
 #define GETNAMEINFO_QUAL_ARG1 const
@@ -108,7 +108,7 @@
 #define HAVE_ALARM 1
 
 /* Define to 1 if you have the <alloca.h> header file. */
-#define HAVE_ALLOCA_H 1
+/* #undef HAVE_ALLOCA_H */
 
 /* Define to 1 if you have the <arpa/inet.h> header file. */
 #define HAVE_ARPA_INET_H 1
@@ -126,7 +126,7 @@
 #define HAVE_BOOL_T 1
 
 /* Define to 1 if you have the clock_gettime function and monotonic timer. */
-/* #undef HAVE_CLOCK_GETTIME_MONOTONIC */
+#define HAVE_CLOCK_GETTIME_MONOTONIC 1
 
 /* Define to 1 if you have the `closesocket' function. */
 /* #undef HAVE_CLOSESOCKET */
@@ -141,7 +141,7 @@
 /* #undef HAVE_DES_H */
 
 /* Define to 1 if you have the <dlfcn.h> header file. */
-#define HAVE_DLFCN_H 1
+/* #undef HAVE_DLFCN_H */
 
 /* Define to 1 if you have the `ENGINE_load_builtin_engines' function. */
 #define HAVE_ENGINE_LOAD_BUILTIN_ENGINES 1
@@ -159,7 +159,7 @@
 #define HAVE_FCNTL_H 1
 
 /* Define to 1 if you have a working fcntl O_NONBLOCK function. */
-#define HAVE_FCNTL_O_NONBLOCK 1
+/* #undef HAVE_FCNTL_O_NONBLOCK */
 
 /* Define to 1 if you have the fdopen function. */
 #define HAVE_FDOPEN 1
@@ -177,7 +177,7 @@
 #define HAVE_FTRUNCATE 1
 
 /* Define to 1 if you have a working getaddrinfo function. */
-#define HAVE_GETADDRINFO 1
+/* #undef HAVE_GETADDRINFO */
 
 /* Define to 1 if you have the `geteuid' function. */
 /* #undef HAVE_GETEUID */
@@ -186,7 +186,7 @@
 #define HAVE_GETHOSTBYADDR 1
 
 /* Define to 1 if you have the gethostbyaddr_r function. */
-#define HAVE_GETHOSTBYADDR_R 1
+/* #undef HAVE_GETHOSTBYADDR_R */
 
 /* gethostbyaddr_r() takes 5 args */
 /* #undef HAVE_GETHOSTBYADDR_R_5 */
@@ -195,10 +195,10 @@
 /* #undef HAVE_GETHOSTBYADDR_R_7 */
 
 /* gethostbyaddr_r() takes 8 args */
-#define HAVE_GETHOSTBYADDR_R_8 1
+/* #undef HAVE_GETHOSTBYADDR_R_8 */
 
 /* Define to 1 if you have the gethostbyname function. */
-#define HAVE_GETHOSTBYNAME 1
+/* #undef HAVE_GETHOSTBYNAME */
 
 /* Define to 1 if you have the gethostbyname_r function. */
 /* #undef HAVE_GETHOSTBYNAME_R */
@@ -325,7 +325,7 @@
 #define HAVE_IOCTL_SIOCGIFADDR 1
 
 /* Define to 1 if you have the <io.h> header file. */
-#define HAVE_IO_H 1
+/* #undef HAVE_IO_H */
 
 /* if you have the Kerberos4 libraries (including -ldes) */
 /* #undef HAVE_KRB4 */
@@ -382,7 +382,7 @@
 #define HAVE_LIBSSL 1
 
 /* if zlib is available */
-#define HAVE_LIBZ 1
+/* #undef HAVE_LIBZ */
 
 /* if your compiler supports LL */
 #define HAVE_LL 1
@@ -397,10 +397,10 @@
 #define HAVE_LONGLONG 1
 
 /* Define to 1 if you have the malloc.h header file. */
-#define HAVE_MALLOC_H 1
+/* #undef HAVE_MALLOC_H */
 
 /* Define to 1 if you have the memory.h header file. */
-#define HAVE_MEMORY_H 1
+/* #undef HAVE_MEMORY_H */
 
 /* Define to 1 if you have the MSG_NOSIGNAL flag. */
 /* #undef HAVE_MSG_NOSIGNAL */
@@ -656,10 +656,10 @@
 #define HAVE_SYS_UTIME_H 1
 
 /* Define to 1 if you have the <termios.h> header file. */
-#define HAVE_TERMIOS_H 1
+/* #undef HAVE_TERMIOS_H */
 
 /* Define to 1 if you have the <termio.h> header file. */
-#define HAVE_TERMIO_H 1
+/* #undef HAVE_TERMIO_H */
 
 /* Define to 1 if you have the <time.h> header file. */
 #define HAVE_TIME_H 1
@@ -719,7 +719,7 @@
 /* #undef HAVE_X509_H */
 
 /* if you have the zlib.h header file */
-#define HAVE_ZLIB_H 1
+/* #undef HAVE_ZLIB_H */
 
 /* Define to 1 if you need the lber.h header file even with ldap.h */
 /* #undef NEED_LBER_H */
@@ -746,19 +746,19 @@
 #define PACKAGE "curl"
 
 /* a suitable file to read random data from */
-#define RANDOM_FILE "/dev/urandom"
+/* #undef RANDOM_FILE */
 
 /* Define to the type of arg 1 for recvfrom. */
 #define RECVFROM_TYPE_ARG1 int
 
 /* Define to the type pointed by arg 2 for recvfrom. */
-#define RECVFROM_TYPE_ARG2 void
+#define RECVFROM_TYPE_ARG2 char
 
 /* Define to 1 if the type pointed by arg 2 for recvfrom is void. */
-#define RECVFROM_TYPE_ARG2_IS_VOID 1
+/* #undef RECVFROM_TYPE_ARG2_IS_VOID */
 
 /* Define to the type of arg 3 for recvfrom. */
-#define RECVFROM_TYPE_ARG3 size_t
+#define RECVFROM_TYPE_ARG3 int
 
 /* Define to the type of arg 4 for recvfrom. */
 #define RECVFROM_TYPE_ARG4 int
@@ -770,7 +770,7 @@
 /* #undef RECVFROM_TYPE_ARG5_IS_VOID */
 
 /* Define to the type pointed by arg 6 for recvfrom. */
-#define RECVFROM_TYPE_ARG6 socklen_t
+#define RECVFROM_TYPE_ARG6 int
 
 /* Define to 1 if the type pointed by arg 6 for recvfrom is void. */
 /* #undef RECVFROM_TYPE_ARG6_IS_VOID */
@@ -782,10 +782,10 @@
 #define RECV_TYPE_ARG1 int
 
 /* Define to the type of arg 2 for recv. */
-#define RECV_TYPE_ARG2 void *
+#define RECV_TYPE_ARG2 char *
 
 /* Define to the type of arg 3 for recv. */
-#define RECV_TYPE_ARG3 size_t
+#define RECV_TYPE_ARG3 int
 
 /* Define to the type of arg 4 for recv. */
 #define RECV_TYPE_ARG4 int
@@ -818,10 +818,10 @@
 #define SEND_TYPE_ARG1 int
 
 /* Define to the type of arg 2 for send. */
-#define SEND_TYPE_ARG2 void *
+#define SEND_TYPE_ARG2 char *
 
 /* Define to the type of arg 3 for send. */
-#define SEND_TYPE_ARG3 size_t
+#define SEND_TYPE_ARG3 int
 
 /* Define to the type of arg 4 for send. */
 #define SEND_TYPE_ARG4 int
@@ -836,7 +836,7 @@
 #define SIZEOF_LONG 4
 
 /* The size of `off_t', as computed by sizeof. */
-#define SIZEOF_OFF_T 8
+#define SIZEOF_OFF_T 4
 
 /* The size of `short', as computed by sizeof. */
 #define SIZEOF_SHORT 2
@@ -878,7 +878,7 @@
 /* #undef USE_NSS */
 
 /* if OpenSSL is in use */
-#define USE_OPENSSL 1
+/* #define USE_OPENSSL 1 */   /* NI-CHANGE: Will be set by MAK file only for SSL-enabled builds. */
 
 /* Define to 1 if you are building a Windows target without large file
    support. */
-- 
2.17.1

